grant/policy.go

```diff
@@ -51,28 +51,27 @@ func (p Policy) IsEmpty() bool {
 // IsDenied returns true if the given license is denied by the policy
 func (p Policy) IsDenied(license License, pkg *Package) (bool, *Rule) {
 	for _, rule := range p.Rules {
-		if rule.Mode != Deny {
-			continue
-		}
-
 		var toMatch string
+
 		if license.IsSPDX() {
 			toMatch = strings.ToLower(license.LicenseID)
 		} else {
 			toMatch = strings.ToLower(license.Name)
 		}
 
 		toMatch = strings.ToLower(toMatch)
+		// TODO: write tests for this section
 		if rule.Glob.Match(toMatch) && toMatch != "" {
 			if pkg == nil {
 				return true, &rule
 			}
 			for _, exception := range rule.Exceptions {
 				if exception.Match(pkg.Name) {
-					return false, &rule
+					return rule.Mode != Deny, &rule
 				}
 			}
-			return true, &rule
+			// true when Mode=Deny, false otherwise
+			return rule.Mode == Deny, &rule
 		}
```

Comment:
Maintainers: what is the expected behavior when `"mode": "ignore"`? Is it supposed to behave the same as `"mode": "allow"`? I couldn't find any documentation or code to understand that mode

Reply:
Thanks! Yes I need to add a description and example for `ignore` - the thought behind it was this.

If you run `grant` then by default it should return 🔴 given that nothing has been `allowed`.

Ignore❔was a way for users to track, by configuration, licenses that we're not saying 🟢 this through, but we don't know what to do with it yet. That gives some interesting future development options for displaying ignored license patterns.

Reply:
I've moved ignore to behave as allow in this PR with the plan for a future PR to build more explicit behavior at the CLI level around the return of an ignore match.